src/client/renderingengine.cpp

```diff
@@ -0,0 +1,1072 @@
+	if (show_hud) {
+		hud->drawSelectionMesh();
+		if (draw_wield_tool)
+			camera->drawWieldedTool(&leftMove);
+	}
+
+	guienv->drawAll();
+
+	// Right eye...
+	irr::core::vector3df rightEye;
+	irr::core::matrix4 rightMove;
+	rightMove.setTranslation(irr::core::vector3df(
+			g_settings->getFloat("3d_paralax_strength"), 0.0f, 0.0f));
+	rightEye = (startMatrix * rightMove).getTranslation();
+
+	// clear the depth buffer
+	getVideoDriver()->clearZBuffer();
+	getVideoDriver()->getOverrideMaterial().Material.ColorMask =
+			irr::video::ECP_GREEN + irr::video::ECP_BLUE;
+	getVideoDriver()->getOverrideMaterial().EnableFlags = irr::video::EMF_COLOR_MASK;
+	getVideoDriver()->getOverrideMaterial().EnablePasses =
+			irr::scene::ESNRP_SKY_BOX + irr::scene::ESNRP_SOLID +
+			irr::scene::ESNRP_TRANSPARENT +
+			irr::scene::ESNRP_TRANSPARENT_EFFECT + irr::scene::ESNRP_SHADOW;
+	camera->getCameraNode()->setPosition(rightEye);
+	camera->getCameraNode()->setTarget(focusPoint);
+	get_scene_manager()->drawAll();
+	getVideoDriver()->setTransform(video::ETS_WORLD, core::IdentityMatrix);
+	if (show_hud) {
+		hud->drawSelectionMesh();
+		if (draw_wield_tool)
+			camera->drawWieldedTool(&rightMove);
+	}
+
+	guienv->drawAll();
+
+	getVideoDriver()->getOverrideMaterial().Material.ColorMask = irr::video::ECP_ALL;
+	getVideoDriver()->getOverrideMaterial().EnableFlags = 0;
+	getVideoDriver()->getOverrideMaterial().EnablePasses = 0;
+	camera->getCameraNode()->setPosition(oldPosition);
+	camera->getCameraNode()->setTarget(oldTarget);
+}
+
+void RenderingEngine::init_texture(
+		const v2u32 &screensize, video::ITexture **texture, const char *name)
+{
+	if (*texture) {
+		getVideoDriver()->removeTexture(*texture);
+	}
+	*texture = getVideoDriver()->addRenderTargetTexture(
+			core::dimension2d<u32>(screensize.X, screensize.Y), name,
+			irr::video::ECF_A8R8G8B8);
+}
+
+video::ITexture *RenderingEngine::draw_image(const v2u32 &screensize, paralax_sign psign,
+		const irr::core::matrix4 &startMatrix,
+		const irr::core::vector3df &focusPoint, bool show_hud, Camera *camera,
+		Hud *hud, bool draw_wield_tool, Client *client,
+		gui::IGUIEnvironment *guienv, const video::SColor &skycolor)
+{
+	static video::ITexture *images[2] = {NULL, NULL};
+	static v2u32 last_screensize = v2u32(0, 0);
+
+	video::ITexture *image = NULL;
+
+	if (screensize != last_screensize) {
+		init_texture(screensize, &images[1], "mt_drawimage_img1");
+		init_texture(screensize, &images[0], "mt_drawimage_img2");
+		last_screensize = screensize;
+	}
+
+	if (psign == RIGHT)
+		image = images[1];
+	else
+		image = images[0];
+
+	getVideoDriver()->setRenderTarget(image, true, true,
+			irr::video::SColor(255, skycolor.getRed(), skycolor.getGreen(),
+					skycolor.getBlue()));
+
+	irr::core::vector3df eye_pos;
+	irr::core::matrix4 movement;
+	movement.setTranslation(irr::core::vector3df(
+			(int)psign * g_settings->getFloat("3d_paralax_strength"), 0.0f,
```

Comment:
I fixed the enum as it's a private enum, but i don't modify the setting name which is out of this PR scope